From c02fc7daf3790e48c07628c5096867b1eb272d77 Mon Sep 17 00:00:00 2001
From: Kenneth Moreland <morelandkd@ornl.gov>
Date: Thu, 27 Aug 2026 14:05:29 -0600
Subject: [PATCH] Update ANARI lighting specification

The ANARI 1.0 specification has an extension named KHR_AREA_LIGHTS that defines
the radius and other parameters of lights that give the lights a physical size
and creates effects soft shadows. However, this extension was removed from ANARI
specification 1.1. The parameters have been moved to the extensions that define
lights (which are free to ignore them if soft shadows are not supported).

The ANARI 1.1 specification was first supported by the ANARI SDK version 0.16.
---
 .../release/dev/anari-light-extensions.md     |  9 +++++
 Rendering/ANARI/vtkAnariLightNode.cxx         | 33 ++++++-------------
 2 files changed, 19 insertions(+), 23 deletions(-)
 create mode 100644 Documentation/release/dev/anari-light-extensions.md

diff --git a/Documentation/release/dev/anari-light-extensions.md b/Documentation/release/dev/anari-light-extensions.md
new file mode 100644
index 00000000000..f50029cfaea
--- /dev/null
+++ b/Documentation/release/dev/anari-light-extensions.md
@@ -0,0 +1,9 @@
+## Updated ANARI to 1.1 specification
+
+The ANARI 1.0 specification has an extension named KHR_AREA_LIGHTS that defines
+the radius and other parameters of lights that give the lights a physical size
+and creates effects soft shadows. However, this extension was removed from ANARI
+specification 1.1. The parameters have been moved to the extensions that define
+lights (which are free to ignore them if soft shadows are not supported).
+
+The ANARI 1.1 specification was first supported by the ANARI SDK version 0.16.
diff --git a/Rendering/ANARI/vtkAnariLightNode.cxx b/Rendering/ANARI/vtkAnariLightNode.cxx
index 15e9af9c245..6cb20ef1e2d 100644
--- a/Rendering/ANARI/vtkAnariLightNode.cxx
+++ b/Rendering/ANARI/vtkAnariLightNode.cxx
@@ -328,18 +328,9 @@ void vtkAnariLightNode::Synchronize(bool prepass)
         anari::setParameter(anariDevice, anariLight, "position", lightPosition);
         // The overall amount of light emitted by the light in a direction in W/sr
         anari::setParameter(anariDevice, anariLight, "intensity", lightIntensity);
-
         // The size of the point light
-        if (anariExtensions.ANARI_KHR_AREA_LIGHTS)
-        {
-          float radius = static_cast<float>(vtkAnariLightNode::GetRadius(light));
-          anari::setParameter(anariDevice, anariLight, "radius", radius);
-        }
-        else
-        {
-          this->Internals->RendererNode->WarningMacroOnce(
-            this, " doesn't support KHR_AREA_LIGHTS::radius");
-        }
+        float radius = static_cast<float>(vtkAnariLightNode::GetRadius(light));
+        anari::setParameter(anariDevice, anariLight, "radius", radius);
       }
       else
       {
@@ -390,17 +381,9 @@ void vtkAnariLightNode::Synchronize(bool prepass)
         static_cast<float>(vtkAnariLightNode::GetLightScale(light) * light->GetIntensity());
       anari::setParameter(anariDevice, anariLight, "irradiance", irradiance);
 
-      if (anariExtensions.ANARI_KHR_AREA_LIGHTS)
-      {
-        float radius = static_cast<float>(vtkAnariLightNode::GetRadius(light));
-        // apparent size (angle in radians) of the light
-        anari::setParameter(anariDevice, anariLight, "angularDiameter", radius);
-      }
-      else
-      {
-        this->Internals->RendererNode->WarningMacroOnce(
-          this, " doesn't support KHR_AREA_LIGHTS::angularDiameter");
-      }
+      float radius = static_cast<float>(vtkAnariLightNode::GetRadius(light));
+      // apparent size (angle in radians) of the light
+      anari::setParameter(anariDevice, anariLight, "angularDiameter", radius);
     }
     else
     {
@@ -414,7 +397,11 @@ void vtkAnariLightNode::Synchronize(bool prepass)
     // All light sources accept the following parameters
     anari::setParameter(anariDevice, anariLight, "color", lightColor);
 
+#if ((ANARI_SDK_VERSION_MAJOR * 100) + ANARI_SDK_VERSION_MINOR) >= 16
+    if (anariExtensions.ANARI_KHR_LIGHT_PRIMARY_VISIBILITY)
+#else
     if (anariExtensions.ANARI_KHR_AREA_LIGHTS)
+#endif
     {
       bool isVisible = light->GetSwitch() ? true : false;
       anari::setParameter(anariDevice, anariLight, "visible", isVisible);
@@ -422,7 +409,7 @@ void vtkAnariLightNode::Synchronize(bool prepass)
     else
     {
       this->Internals->RendererNode->WarningMacroOnce(
-        this, " doesn't support KHR_AREA_LIGHTS::visible");
+        this, " doesn't support KHR_LIGHT_PRIMARY_VISIBILITY::visible");
     }
 
     anari::commitParameters(anariDevice, anariLight);
-- 
GitLab

